lib/hashtable.hh

```diff
@@ -110,6 +113,17 @@ protected:
     explicit Hashtable(const Hashtable&);
     Hashtable& operator=(const Hashtable&);
 
+    virtual unique_ptr<KmerHashIterator>
```

Comment:
I wonder if code clarity would benefit from typedef'ing `std::unique_ptr<KmerHashIterator>` to something more concise. The counterargument would be that it's not as explicit, but concise code would be a bigger win IMO.